api/apps/api/src/modules/geo-features/geo-features.service.ts

```diff
@@ -431,6 +452,24 @@ export class GeoFeaturesService extends AppBaseService<
     return right(geoFeature);
   }
 
+  private async saveFeatureAmountPerPlanningUnit(
+    geoEntityManager: EntityManager,
+    projectId: string,
+    featureAmounts: ComputeFeatureAmountPerPlanningUnit[],
+  ): Promise<void> {
+    const repo = geoEntityManager.getRepository(
+      FeatureAmountsPerPlanningUnitEntity,
+    );
+    await repo.save(
+      featureAmounts.map(({ amount, projectPuId, featureId }) => ({
+        projectId,
+        featureId,
+        amount,
+        projectPuId,
+      })),
+    );
+  }
```

Comment:
I think this may break under pressure, aka when inserting a high enough number of PU amounts at once so that the combined number of query parameters goes above PostgreSQL's limit for this.


```suggestion
// add to imports
import { CHUNK_SIZE_FOR_BATCH_GEODB_OPERATIONS } from '@marxan-geoprocessing/utils/chunk-size-for-batch-geodb-operations';
//and here
    await repo.save(
      featureAmounts.map(({ amount, projectPuId, featureId }) => ({
        projectId,
        featureId,
        amount,
        projectPuId,
      })),
      {
        chunk: CHUNK_SIZE_FOR_BATCH_GEODB_OPERATIONS,
      }
    );
  }
```

Reply:
🤔 I thought I had posted a follow-up comment via my IDE but I think I may have messed that up - basically it was about my ill-informed advice to use `CHUNK_SIZE_FOR_BATCH_GEODB_OPERATIONS` as this is defined in the geoprocessing service and so it will not be importable directly in the api service. I had some random rants about semantics in my vanished comment, but basically tl;dr please let's use `CHUNK_SIZE_FOR_BATCH_APIDB_OPERATIONS` here without moving anything else around (moving constants to libs, etc.)